tfc-dump.pl

```diff
@@ -115,29 +115,39 @@
 my @vs_names;
 my @vs_ids;
 my $pg_size = 100;
+my $pg_num = 1;
 my $total_count;
 my $tmpfile = `mktemp`;
 chomp($tmpfile);
 
-$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/varsets?page%5Bsize%5D=${pg_size}\"";
-$curl_cmd   = "curl $curl_headers --output $tmpfile $curl_query";
-system($curl_cmd);
-
-# Get the Variable Set names
-
-$jq_cmd   = "cat $tmpfile | jq '.data[].attributes.name'";
-@vs_names = `$jq_cmd`;
-# Remove the double quotes in each element of the array.
-grep($_ =~ s/"//g && 0, @vs_names);	# Programming Perl, p. 221, 1990
-chomp(@vs_names);
-
-# Get the Variable Set IDs
+do {
+	$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/varsets?page%5Bsize%5D=${pg_size}&page%5number%5D=${pg_num}\"";
+	$curl_cmd   = "curl $curl_headers --output $tmpfile $curl_query";
+	system($curl_cmd);
 
-$jq_cmd = "cat $tmpfile | jq '.data[].id'";
-@vs_ids = `$jq_cmd`;
-# Remove the double quotes in each element of the array.
-grep($_ =~ s/"//g && 0, @vs_ids);	# Programming Perl, p. 221, 1990
-chomp(@vs_ids);
+	# Get the Variable Set names
+	$jq_cmd   = "cat $tmpfile | jq '.data[].attributes.name'";
+	my @page_names = `$jq_cmd`;
+
+	# Remove the double quotes in each element of the array.
+	grep($_ =~ s/"//g && 0, @page_names);	# Programming Perl, p. 221, 1990
+	chomp(@page_names);
+	push(@vs_names, @page_names);
+
+	# Get the Variable Set IDs from this page
+    $jq_cmd = "cat $tmpfile | jq '.data[].id'";
+    my @page_ids = `$jq_cmd`;
+    grep($_ =~ s/"//g && 0, @page_ids);
+    chomp(@page_ids);
+    push(@vs_ids, @page_ids);
+
+	# Get pagination metadata
+    $jq_cmd = "cat $tmpfile | jq '.meta.pagination[\"total-count\"]'";
+    $total_count = `$jq_cmd`;
```

Comment:
using system on line 126 is inconsistent with the use of back ticks here. system is preferred.

Reply:
backticks don't catch errors as well.

#/usr/bin/perl

# Modern equivalent to use strict; use warnings;
use Modern::Perl;
# To allow more readable variable names (e.g. $OS_ERROR instead of $!)
use English qw( -no_match_vars );

# single quotes preferred, unless glob'bing
my $command = 'ls -la';

# Open a pipe to capture the output of the $command, $OS_ERROR is glob'd.
open(my $fh, '-|', $command) || die "Failed to execute command: $OS_ERROR";
# slurp all the output into an array
my @results_system = <$fh>;
close($fh);
# Make is a string.
$result_system = join qq{}, @results_system;

Reply:
After further thought, fixing the backticks is more effort than necessary. I swung to the "perlcritic -2" level of suggestions. No need.